src/components/ActivityForm.js

```diff
@@ -27,57 +27,59 @@ export default class ActivityForm extends React.Component {
       name: props.activity ? props.activity.name : '',
       room: props.activity ? props.activity.room : '',
       teacher: props.activity ? props.activity.teacher : '',
-      error: '',
+      classNoError: '',
+      nameError: '',
+      roomError: '',
+      teacherError: '',
       // TODO: when you will use prop-types you will be able to improve this place ;)
       // TODO: I think, that now (looking at previous comments, maybe you will be able to
       // improve). If not, because activity will be able to be undefined, in separate PR
       // you should check lodash package and get method ;)
     };
   }
 
-  onNameChange = e => this.setState({ name: e.target.value });
-
-  onTeacherChange = e => this.setState({ teacher: e.target.value });
-
-  onRoomChange = e => this.setState({ room: e.target.value });
-
-  onClassNoChange = e => {
-    const classNo = e.target.value;
-    if (!classNo || classNo.match(/^[1-9]\d*$/)) {
-      this.setState({ classNo });
+  onInputValueChange = field => e => {
+    const inputValue = e.target.value;
+    if (field === 'classNo' && (!inputValue || inputValue.match(/^[1-9]\d*$/))) {
```

Comment:
I understand that, class number should be a number. I am very happy, that you learned about regex :D 
But we have better options for that. 
We should create some helper (good approach is to have `helpers` folder in src, for validators, we could have `validators` file). We should use `isNaN` built in Number method. 
Btw...for checking if value is is not empty, we could also create a function there where we could use also `trim` built in String method to be sure, that there are no empty spaces in the string. 
---
Validations are really, really important. Maybe this is a good time, to read and check `lodash` package :) It's up to you, but Lodash is great utility library.

Reply:
also ,we need to remember, that you are not able to write letters (except `e`) in input with type number ;)